app/goal_routes.py

```diff
@@ -0,0 +1,106 @@
+from flask import Blueprint, json, jsonify, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+from datetime import datetime
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if 'title' not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    new_goal = Goal(
+        title=request_body['title']
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def create_tasks_for_goal(goal_id):
+    request_body = request.get_json()
+
+    goal = Goal.query.get(goal_id)
+
+    if not goal:
+        return "", 404
+
+    for id in request_body['task_ids']:
+        goal.tasks.append(Task.query.get(id))
+
+    db.session.commit()
+
+    return {
+        "id": goal.goal_id,
+        "task_ids": request_body['task_ids']
+    }, 200
+
+@goals_bp.route("", methods=["GET"])
+def read_goals():
+    goals = Goal.query.all()
+
+    goals_response = []
+
+    if not goals:
+        return jsonify(goals_response), 200
+
+    for goal in goals:
+        goals_response.append(goal.to_dict())
```

Comment:
The if-block is unneeded here. If there are no goals, the for loop will be skipped and the response will still be appropriately empty.